File: Source/Service/appService/com.maxiaohua.genealogy.main.biz/src/com/maxiaohua/genealogy/main/biz/a01/dto/PhoneNoteDTO.java
/*
 * Genealogy
 * Xiao Hua Ma personal studio 2015.
 */
package com.maxiaohua.genealogy.main.biz.a01.dto;

/**
 * PhoneNoteDTO DTO
 *
 * History
 * REV.         Updated Date           Updater              Infomation
 * -------      ---------------        ----------------     ------------------
 * 1.0          2015/09/06             TOOL                Create
*
 */
@com.maxiaohua.genealogy.fw.core.type.AliasKanJi("电话本信息")
public class PhoneNoteDTO implements java.io.Serializable, Comparable<PhoneNoteDTO> {
	/**
	* System ID
	*/
	private static final long serialVersionUID = 1L;

	private String name = null;

	public String getName() {
		return this.name;
	}

	public void setName(
			String name) {
		this.name = name;
	}

	private String phone = null;

	public String getPhone() {
		return this.phone;
	}

	public void setPhone(
			String phone) {
		this.phone = phone;
	}

  	/* (non-Javadoc)
     * @see java.lang.Comparable#compareTo(java.lang.Object)
     */
    @Override
    public int compareTo(
    		PhoneNoteDTO compareToDTO) {
        return this.equals(compareToDTO) ? 0 : -1;
    }
}

File: Source/iOS/Genealogy/Genealogy/Model/A01/DTO/PhoneNoteDTO.m
//
//  PhoneNoteDTO.m
//  Genealogy
//
//  Created by xiao huama on 15/09/10.
//  Xiao Hua Ma personal studio 2015.
//

#import "PhoneNoteDTO.h"

@implementation PhoneNoteDTO

- (void)loadDataWithJsonData:(id)jsonData
{
    if (![jsonData isKindOfClass:[NSDictionary class]]) {
        return;
    }
    if (!jsonData) {
        return;
    }
    if (![jsonData[@"name"] isKindOfClass:[NSNull class]]) {
        self.name = jsonData[@"name"];
    }
    if (![jsonData[@"phone"] isKindOfClass:[NSNull class]]) {
        self.phone = jsonData[@"phone"];
    }
}

-(NSMutableDictionary *)buildRequestData
{
    NSMutableDictionary *dic = [NSMutableDictionary dictionary];
    if (self.name) {
        dic[@"name"] = self.name;
    }
    if (self.phone) {
        dic[@"phone"] = self.phone;
    }
    return dic;
}

@end
